avocado/core/remoter.py

```diff
@@ -78,7 +87,9 @@ def __init__(self, hostname, username=None, password=None,
                                 connection_attempts=attempts,
                                 linewise=True,
                                 abort_on_prompts=True,
-                                abort_exception=ConnectionError)
+                                abort_exception=ConnectionError,
+                                reject_unknown_hosts=reject_unknown_hosts,
```

Comment:
Well I know it's a known issue, but when using ECDSA (so from paramiko point of view invalid) key the message is really misleading:

```
avocado run --remote-hostname localhost passtest
LOGIN      : medic@localhost:22 (TIMEOUT: 60 seconds)

Avocado job failed: JobError: Needed to prompt for a connection or sudo password (host: localhost), but abort-on-prompts was set to True
```

I tried to look at the original fabric exception and it does not include any hits on why (even without the custom `abort_exception` it only contains this message). I went ahead and created https://github.com/fabric/fabric/pull/1458 to fix this, so hopefully new versions will be friendlier.

We could catch the exception and try adding more-generic exception, but as fabric masks the internal information about the exception I think we better rely on it and not try to be smart.